### DISCUSSION TITLE
Saving mitsuba 3 scene

### DISCUSSION THREAD

**Discussion by sapo17** [mitsuba-renderer/mitsuba3#388] [Q&A] [answered]
Hello,

sorry if a similar question is already answered, but I could not find a similar topic anywhere.

So in the mitsuba 3 [documentation](https://mitsuba.readthedocs.io/en/stable/src/api_reference.html?highlight=update_scene#mitsuba.load_file) the method  `mitsuba.load_file(path, update_scene=False, parallel=False, **kwargs)` has a parameter `update_scene`, which documentation indicates wheter the updated scene contents will be written back to disk.

So, really naively I am trying the following to see whether the mitsuba updates scene to a newer version and consequently the updates written back to disk:

```
import mitsuba as mi
mi.set_variant('cuda_ad_rgb')
scene = mi.load_file('scenes/editing-a-scene-mitsuba-tutorial/cbox.xml', update_scene=True)
params = mi.traverse(scene)
print(params['red.reflectance.value']) # prints [[0.5700680017471313, 0.043013498187065125, 0.04437059909105301]]
params['red.reflectance.value'] = mi.Color3f(0.0, 0.0, 0.0)
params.update() # expecting the updates to written back to the original xml file
print(params['red.reflectance.value']) # prints [[0.0, 0.0, 0.0]]
```

Unfortunately the updated scene is not written back to disk.

Am I missing something, or that is not the use case for `update_scene`?

Is there a way to export a modified mitsuba scene (e.g. after optimization finished)?

Thanks in advance!
Can

**Comment by Speierers** [comment, marked as answer]
Hi @sapo17 ,

No unfortunately this is not what the `update_scene` does in `load_file`. It can be used to update an old Mitsuba XML file (e.g. one written for Mitsuba 0.7 or Mitsuba 2). For instance Mitsuba 2 and 3 uses `snake_case` convention in parameter names while Mitsuba 0.7 used `CamelCase`.

There is no good way to write back the updates done to a scene instance to XML. Also the scene object parameters (in `params`) don't map one-to-one with the XML parameters, so this will be difficult.

**Reply by sapo17** [reply]
Thank you for the clarification!
